Feat/#103 add update password

Teodor25 opened the pull request:
- Fix verify issue where it doesnt update isVerified for navbar and profile 
- add newsletter signup logic to signup and checkout
- add unsubscribe from newsletter page
- add change password to profile page
- update home page styling 
- update profile validators 
- update localisation 
- add e2e tests for profile 
- add e2e tests for unsubscribe page 

Closes: #103 #102 #101

Teodor25 committed:
fix: 🐛 Update profile page validators and error messages

Teodor25 committed:
fix: 🐛 Update homepage styling and add change password modal

Teodor25 committed:
feat: 🎸 Minor styling updates

Update cookie prompt icon, update review section styling on mobile and
desktop, add the changePasswordModal to profile, update profile styling
for better responsiveness

Teodor25 committed:
feat: 🎸 Add change password modal

update localisation, add change password modal, localise change
password, update e2e profile, update login button error when loading,
fix update user with new email error and add new password required
description

Teodor25 committed:
fix: 🐛 Add localisation destination folder

Teodor25 committed:
feat: 🎸 Add update password functionality and newsletter signup

Teodor25 committed:
feat: 🎸 Add newsletter signup logic to checkout

Teodor25 committed:
fix: 🐛 Fix verify service error not updating isVerified

Created unsubscribe page, updated localization, fix error in
verification

Kwandes commented on apps/webstore/src/app/app-routing.module.ts:
```diff
@@ -31,6 +32,11 @@ const routes: Routes = [
+    path: 'unsubscribe/:token',
+    component: UnsubscribeComponent,
+    canActivate: [CookieGuard],
+  },
```
Should be `newsletter/unsubscribe:newsletterId` to separate it from other functionality
Keep in mind that this is a `newsletterId`, not a `token`, so name it accordingly
_warning: untested code_
```suggestion
  {
    path: 'newsletter',
    children: [
      {
        path: 'unsubscribe/:newsletterId',
        component: UnsubscribeComponent,
        canActivate: [CookieGuard],
      },
    ],
  },
```

Kwandes commented on apps/webstore/src/app/pages/auth/checkout/checkout.component.ts:
```diff
@@ -174,6 +175,32 @@ export class CheckoutComponent implements OnInit {
 
+  subscribeUserToNewsletter() {
+    //TODO: add a fucntion that automatically sends an email to the user 3 weekes
+    //after the purchase is completed with a discount code that works once for 25% off
```
Remove this TODO, it is only relevant in the basket regarding the purchase
```suggestion
```

Kwandes commented on apps/webstore/src/app/pages/auth/checkout/checkout.component.ts:
```diff
@@ -188,9 +215,10 @@ export class CheckoutComponent implements OnInit {
+      this.subscribeUserToNewsletter();
     }
 
+    //TODO: create a Quickpay call and redirect to a payment link
```
Remove this TODO, it is only relevant in the basket regarding the purchase
```suggestion
```

Kwandes commented on apps/webstore/src/app/pages/auth/profile/profile.component.html:
```diff
@@ -144,7 +144,7 @@ <h2 i18n="@@account-info">Account info</h2>
                 data-cy="account-invalid-address"
             >
-                The address is too long.
+                Invalid address
```
The alert should say why it is invalid
```suggestion
                The address is too long
```

Kwandes commented on apps/webstore/src/app/pages/auth/profile/profile.component.html:
```diff
@@ -171,7 +171,7 @@ <h2 i18n="@@account-info">Account info</h2>
                 data-cy="account-invalid-address-2"
             >
-                The address is too long.
+                Invalid address
```
The alert should say why it is invalid
```suggestion
                The address is too long
```

Kwandes commented on apps/webstore/src/app/pages/auth/profile/profile.component.html:
```diff
@@ -197,7 +197,7 @@ <h2 i18n="@@account-info">Account info</h2>
                         data-cy="account-invalid-city"
                     >
-                        The city name is too long.
+                        Invalid city
```
The alert should say why it is invalid
```suggestion
                The city name is too long
```

Kwandes commented on apps/webstore/src/app/pages/auth/profile/profile.component.ts:
```diff
@@ -45,25 +48,10 @@ export class ProfileComponent implements OnInit {
       name: new FormControl('', [
         Validators.maxLength(50),
-        Validators.pattern("^[a-zA-Z-' ]*$"),
+        Validators.pattern("^[a-zA-Z-' ÆØÅæøå]*$"),
```
We shouldn't verify that the names contain valid characters etc
It is way too much work to handle all possibilities (Do you want to add EVERY SPECIAL CHARACTER out there? Example: ´ąółżęć´ just for Poland)
```suggestion
```
This comment also applies to the city and address

Kwandes commented on apps/webstore/src/app/pages/auth/profile/profile.component.ts:
```diff
@@ -90,6 +78,20 @@ export class ProfileComponent implements OnInit {
+      });
+    } catch (error) {
+      console.error(error);
+      // TODO: handle failed fetching of the data
```
Address this TODO (add an alert on the page, not just a toast message)

Kwandes commented on apps/webstore/src/app/pages/auth/signup/signup.component.html:
```diff
@@ -49,17 +49,20 @@ <h1 class="page-title" i18n="@@signup">Sign up</h1>
         </div>
-
+        <div class="info-message" i18n="@@info-password">
+            Password must contain: min 9 characters, at least one upper case
```
The password minimum length is 6
```suggestion
            Password must contain: min 6 characters, at least one upper case
```

Kwandes commented on apps/webstore/src/app/pages/auth/unsubscribe/unsubscribe.component.html:
```diff
@@ -0,0 +1,24 @@
+        </p>
+    </div>
+    <div *ngIf="!isLoading && !unsubscribeSuccessful" class="failed">
+        <h3 i18n="@@unsubscribe-failed">Unsubscribe failed</h3>
```
Should specify the reason for failure, based on the API response
So invalid token, never subscribed, etc
Don't just parse straight from the response, instead read the status and adjust the alert accordingly only for specific cases (don't say "500 internal Server Error" etc)

You can find an example in the verification pages

Kwandes commented on apps/webstore/src/app/shared/components/modals/change-password-modal/change-password-modal.component.html:
```diff
@@ -0,0 +1,113 @@
+                        >
+                    </div>
+                    <div class="info-message" i18n="@@info-password">
+                        Password must contain: min 9 characters, at least one
```
The minimum password length is 6 characters
```suggestion
                        Password must contain: min 6 characters, at least one
```

Kwandes reviewed (changes requested):
There is quite a few issues, please resolve them and re-request a review

Teodor25 committed:
Update apps/webstore/src/app/app-routing.module.ts

Co-authored-by: Kwandes <22862227+Kwandes@users.noreply.github.com>

Teodor25 committed:
Update apps/webstore/src/app/pages/auth/checkout/checkout.component.ts

Co-authored-by: Kwandes <22862227+Kwandes@users.noreply.github.com>

Teodor25 committed:
Update apps/webstore/src/app/pages/auth/checkout/checkout.component.ts

Co-authored-by: Kwandes <22862227+Kwandes@users.noreply.github.com>

Teodor25 committed:
Update apps/webstore/src/app/pages/auth/profile/profile.component.html

Co-authored-by: Kwandes <22862227+Kwandes@users.noreply.github.com>

Teodor25 committed:
Update apps/webstore/src/app/pages/auth/profile/profile.component.html

Co-authored-by: Kwandes <22862227+Kwandes@users.noreply.github.com>

Teodor25 committed:
Update apps/webstore/src/app/pages/auth/profile/profile.component.ts

Co-authored-by: Kwandes <22862227+Kwandes@users.noreply.github.com>

Teodor25 committed:
Update apps/webstore/src/app/shared/components/modals/change-password-modal/change-password-modal.component.html

Co-authored-by: Kwandes <22862227+Kwandes@users.noreply.github.com>

Teodor25 committed:
Update apps/webstore/src/app/pages/auth/signup/signup.component.html

Co-authored-by: Kwandes <22862227+Kwandes@users.noreply.github.com>

Teodor25 committed:
Update apps/webstore/src/app/pages/auth/profile/profile.component.html

Co-authored-by: Kwandes <22862227+Kwandes@users.noreply.github.com>

Teodor25 committed:
fix: 🐛 Fix github comment issues from Calli

Teodor25 committed:
fix: 🐛 Remove google analytics

Kwandes committed:
refactor: 💡 Refactor how the newsletter unsubscribe page looks

Kwandes committed:
refactor: 💡 Remove overly complex city name rule

We don't want to validate for every single special character out in the world...

Kwandes reviewed (changes requested):
I have updated the newsletter unsubscribe page to 1. Properly display feedback, and 2. look like the verification page

But in the process, the localization stuff got changed (since the needed messages are different)
Please update them (there is a todo in `unsubscribe.component.html`)

Also, please include E2E tests for the verification OR make an issue for them to add later

In terms of update password, you were supposed to write the service call for it
You can use the normal `updateUser` method, the DTO includes the password.
And please include E2E tests for it

Teodor25 committed:
feat: 🎸 Update localisation + add styling and icons to unsub

Teodor25 committed:
test: 💍 Add e2e tests for unsubscribe page

Teodor25 committed:
test: 💍 Add e2e tests to profile page change password

Kwandes committed:
refactor: 💡 Adjust error message in unsubscribe page

Kwandes committed:
Merge branch 'development' into feat/#103-add-update-password

Kwandes committed:
refactor: 💡 Change regex for password validation in update pass

Kwandes committed:
refactor: 💡 Change regex for password validation in reset pass

Kwandes committed:
refactor: 💡 Change regex for password validation in signup
